classes/profile.php

```diff
@@ -191,6 +191,12 @@ public function save_record(): int {
         $this->raw_set('timemodified', $now);
         $this->raw_set('usermodified', $USER->id);
 
+        // We may not have obtained a valid userid when the profile record was created.
+        // If the current profile userid is 0, but there's a valid $USER->id, update it.
+        if ((! (int) $this->raw_get('userid')) && (int) $USER->id) {
+            $this->raw_set('userid', (int) $USER->id);
+        }
```

Comment:
There's a lot of casting here :magic_wand: 

Are these necessary? And if not, can we please remove them as they add extra cognitive overhead. I think the user->id check does not need it. Also in the raw_get check for userid, it always returns a value or null if it's not set. I believe an isset check here will be sufficient.